### PR TITLE
Fix/app state

### LINKED ISSUE #1152: Start vpn-sever after the actual app starts
**Describe the bug**
Currently, on `arch-dev` the `vpn-server` takes a while to start because it sometimes takes a while to execute the following command on Linux systems. 
```
iptables -L | grep "Chain FORWARD" | tr -d '()' | awk '{print $4}'
``` 
This currently means that the `vpn-server` is shown as green in the UI and also has its entry in service discovery. This is not a major issue if the above command completes its execution in a couple of seconds. But there has been a case where it took a couple of minutes for the command to execute.

So we need to add an entry in the SD and well as show yellow in UI till the time the `vpn-server` is properly started.

### LINKED ISSUE #1209: Stopping an app gives error
**Describe the bug**
Stopping an app gives the error `No process of given app is running.` but logs show the app closed 

**Environment information:**

- OS: e.g. Linux
- Platform: Linux 5.13.0-41-generic x86_64

**Steps to Reproduce**

1. Start hypervisor
2. Stop Skysocks
3. See error

### PR DESCRIPTION
**Pull request by ersonp**
Did you run `make format && make check`?
yes

Fixes #1209 
Fixes #1152 

 Changes:	
- Added start waitgroup to Proc
- Added app-status to vpn-server, skychat, skysocks client and server 
- Updated methods appCtx, getApp and putApp of Hypervisor

How to test this PR:
1. Add the line `time.Sleep(time.Second*10)` before [this](https://github.com/ersonp/skywire/blob/cdbf44d379dd97ac115389df3cd9b1045624a69b/internal/vpn/server.go#L146)
2. Build `make build`
3. Start hypervisor `./skywire-visor -c skywire-config.json`
4. In the UI stop and restart the `skysocks` or `skychat` app and check the status
5. Start the `vpn-server` (indicator should only turn green after 10 seconds and also the entry to https://sd.skywire.dev/api/services?type=vpn should also be added after 10 seconds)

### CODE REVIEW

**Comment by mrpalide** [comment]
Everything for VPN-Server is OK on develop branch, but in your branch, no entry added to SD unfortunately.
This is visor log:
```
[2022-05-19T06:14:28+04:30] INFO [visor]: Setting app detailed status Connecting for app vpn-server
2022/05/19 06:14:28 [machine/nodASBTldA-000002] "PUT http://localhost:8000/api/visors/025e830f5a9e2f2ecd7884bd6caf74efd8ad5eb98844b753fdaa9768ce30843528/apps/vpn-server HTTP/1.1" from 127.0.0.1:35288 - 200 93B in 17.266894ms
[2022-05-19T06:14:28+04:30] INFO [proc_manager]: Accepted proc conn. hello="{"proc_key":"4fd6809be0204f3792d1f5d8c1cca075"}" remote=127.0.0.1:47374
[2022-05-19T06:14:28+04:30] INFO [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: Associated and serving proc conn.
[2022-05-19T06:14:28+04:30] INFO [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: Request processed. _elapsed="865.081µs" _method="Listen" _received="6:14AM" input=025e830f5a9e2f2ecd7884bd6caf74efd8ad5eb98844b753fdaa9768ce30843528:44 output=0xc00036b78e
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Got app listener, bound to 44"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Got default network interface: wlan0"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Got IPs of interface wlan0: [192.168.1.114]"
[2022-05-19T06:14:28+04:30] INFO [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: Request processed. _elapsed="4.819µs" _method="SetDetailedStatus" _received="6:14AM" input=0xc0000724d0
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Old IP forwarding values:"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="IPv4: 1, IPv6: 1"
[2022-05-19T06:14:28+04:30] INFO (STDOUT) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: net.ipv4.ip_forward = 1
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Old iptables forward policy: ACCEPT"
[2022-05-19T06:14:28+04:30] INFO [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: Request processed. _elapsed="10.849µs" _method="SetDetailedStatus" _received="6:14AM" input=0xc00011d020
[2022-05-19T06:14:28+04:30] INFO (STDOUT) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: net.ipv6.conf.all.forwarding = 1
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Set IPv4 forwarding = 1"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Set IPv6 forwarding = 1"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Enabled IP masquerading"
[2022-05-19T06:14:28+04:30] INFO (STDERR) [proc:vpn-server:4fd6809be0204f3792d1f5d8c1cca075]: time="2022-05-19T06:14:28+04:30" level=info msg="Set iptables forward policy to ACCEPT"

```
This is SD log:
```
[2022-05-19T01:41:45Z] INFO [service-discovery]: geo={Lat:35.698 Lon:51.4115 Country:IR Region:}
2022/05/19 01:41:45 "POST http://sd.skywire.dev/api/services HTTP/1.1" from 188.212.246.142 - 200 181B in 97.62702ms
2022/05/19 01:41:45 "DELETE http://sd.skywire.dev/api/services/025e830f5a9e2f2ecd7884bd6caf74efd8ad5eb98844b753fdaa9768ce30843528:44?type=vpn HTTP/1.1" from 188.212.246.142 - 200 5B in 15.074826ms
```
After posting, a delete request sent immediately too.

File: cmd/apps/vpn-server/vpn-server.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func main() {
 		Secure:           *secure,
 		NetworkInterface: *networkIfc,
 	}
-	srv, err := vpn.NewServer(srvCfg, log)
+	srv, err := vpn.NewServer(srvCfg, log, appClient)
```
Comment (mrpalide) [inline cmd/apps/vpn-server/vpn-server.go]: Could this `appClient` be same in all files. In others is `appC`.

**Review by mrpalide — APPROVED** [review]
Work well, just a suggestion for naming a variables.

File: pkg/visor/hypervisor.go [hunk 2]
```diff
@@ -594,12 +594,7 @@ func (hv *Hypervisor) getApps() http.HandlerFunc {
 // returns an app summary of a given visor's pk and app name
 func (hv *Hypervisor) getApp() http.HandlerFunc {
 	return hv.withCtx(hv.appCtx, func(w http.ResponseWriter, r *http.Request, ctx *httpCtx) {
-		app, err := ctx.API.App(ctx.App.Name)
```
Comment (jdknives) [inline pkg/visor/hypervisor.go:597]: Why are we getting rid of this error check?

Comment (ersonp) [reply pkg/visor/hypervisor.go:597]: I had added it in the previous PR thinking we were not sending the latest App-status. I realized it afterward and fixed that. We do get the latest App-status [here](https://github.com/ersonp/skywire/blob/66cee3c1d43bfcea23ac47e0862b12ed25f7c63c/pkg/visor/hypervisor.go#L1492).

**Pull request merged**